TST: FAILED scipy/interpolate/tests/test_bary_rational.py::TestAAA::test_spiral_cleanup

### Describe your issue.

There's a failing test in `interpolate` when I locally run tests at HEAD.

clang 17 with macOS 15.6.1/xcode26.0.1

### Reproducing Code Example

```python
spin build --with-accelerate
spin test
```

### Error message

```shell
__________________________________________________________________________ TestAAA.test_spiral_cleanup __________________________________________________________________________

self = <scipy.interpolate.tests.test_bary_rational.TestAAA object at 0x131e4df10>

    def test_spiral_cleanup(self):
        z = np.exp(np.linspace(-0.5, 0.5 + 15j*np.pi, num=1000))
        # here we set `rtol=0` to force froissart doublets, without cleanup there
        # are many spurious poles
        with pytest.warns(RuntimeWarning):
            r = AAA(z, np.tan(np.pi*z/2), rtol=0, max_terms=60, clean_up=False)
        n_spurious = np.sum(np.abs(r.residues()) < 1e-14)
        with pytest.warns(RuntimeWarning):
            assert r.clean_up() >= 1
        # check there are less potentially spurious poles than before
        assert np.sum(np.abs(r.residues()) < 1e-14) < n_spurious
        # check accuracy
>       assert_allclose(r(z), np.tan(np.pi*z/2), atol=6e-12, rtol=3e-12)
E       AssertionError: 
E       Not equal to tolerance rtol=3e-12, atol=6e-12
E       
E       nan location mismatch:
E        ACTUAL: array([ 1.406500e+00+3.002888e-14j,  1.397761e+00+1.332687e-01j,
E               1.366773e+00+2.612514e-01j,  1.316442e+00+3.785434e-01j,
E               1.251135e+00+4.813557e-01j,  1.175781e+00+5.678005e-01j,...
E        DESIRED: array([ 1.406500e+00+0.000000e+00j,  1.397761e+00+1.332687e-01j,
E               1.366773e+00+2.612514e-01j,  1.316442e+00+3.785434e-01j,
E               1.251135e+00+4.813557e-01j,  1.175781e+00+5.678005e-01j,...

n_spurious = np.int64(38)
r          = <scipy.interpolate._bary_rational.AAA object at 0x310db3890>
self       = <scipy.interpolate.tests.test_bary_rational.TestAAA object at 0x131e4df10>
z          = array([ 0.60653066+0.00000000e+00j,  0.60646275+2.86287286e-02j,
        0.60504355+5.72510466e-02j,  0.60227352+8.580...273459 +2.31840580e-01j, -1.63810676+1.55002604e-01j,
       -1.64523961+7.76653108e-02j, -1.64872127+8.88608645e-15j])

scipy/interpolate/tests/test_bary_rational.py:260: AssertionError
```

### SciPy/NumPy/Python version and system information

```shell
2.3.3 sys.version_info(major=3, minor=12, micro=6, releaselevel='final', serial=0)
```

--- a/scipy/interpolate/_bary_rational.py
+++ b/scipy/interpolate/_bary_rational.py
@@ -110,7 +110,12 @@ class _BarycentricRational:
         with np.errstate(invalid="ignore", divide="ignore"):
             CC = 1 / np.subtract.outer(zv, self._support_points)
             # Vector of values
-            r = CC @ (weights * support_values) / (CC @ weights)
+            # NOTE: optimized matrix multiplications are avoided here
+            # to prevent the platform-specific deviations noted in
+            # gh-23707
+            num = np.einsum('ik,kj->ij', CC, weights * support_values)
+            denom = np.einsum('ik,kj->ij', CC, weights)
+            r = num / denom
 
         # Deal with input inf: `r(inf) = lim r(z) = sum(w*f) / sum(w)`
         if np.any(np.isinf(zv)):

